dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemoryAllocation.java

```diff
@@ -53,12 +53,38 @@ private int getSize(Malloc malloc) {
         }
     }
 
-    private void createInits(Program program, List<MemoryObject> memoryObjects) {
+    public void moveAndAlignMemoryObjects(Memory memory) {
+        // Addresses are typically at least two byte aligned
+        //      https://stackoverflow.com/questions/23315939/why-2-lsbs-of-32-bit-arm-instruction-address-not-used
+        // Many algorithms rely on this assumption for correctness.
+        // Many objects have even stricter alignment requirements and need up to 8-byte alignment.
+        final BigInteger alignment = BigInteger.valueOf(8);
+        BigInteger nextAddr = alignment;
+        for(MemoryObject memObj : memory.getObjects()) {
+            memObj.setAddress(nextAddr);
+
+            // nextAddr += memObjSize + ((-memObjSize) mod alignment)
```

Comment:
```
nextAddr + size + padding = k*alignment   // Alignment requirement
=> padding = k*alignment - nextAddr - size 
=> padding mod alignment = (-size) mod alignment // k*alignment and nextAddr are 0 mod alignment.
=> padding = (-size) mod alignment // Because padding < alignment
```

Reply:
can you add this to the comment? Otherwise that equation looks like black magic to me